cisc/certificate.go

```diff
@@ -0,0 +1,321 @@
+	return true
+
+}
+func renewCert(cert string) string {
+
+	oldcert, err := pemToCertificate(cert)
+	if err != nil {
+		log.Fatal(err)
+	}
+	//get the serial number and compute the URI
+	serial := fmt.Sprintf("%x", oldcert.SerialNumber)
+	certuri := "https://acme-v01.api.letsencrypt.org/acme/cert/0" + serial
+	// Create a client to the acme of letsencrypt.
+	cli, err := letsencrypt.NewClient("https://acme-v01.api.letsencrypt.org/directory")
+	if err != nil {
+		log.Fatal(err)
+	}
+	//renew the certificate
+	certRen, err := cli.RenewCertificate(string(certuri))
+	if err != nil {
+		log.Fatal("Problem while renewing Certificate:" + err.Error())
+	}
+	//write in the memory
+	fullchain, err := cli.Bundle(certRen)
+	if err != nil {
+		log.Fatal("can't bundles the certificate", err)
+	}
+	ioutil.WriteFile("fullchain.pem", fullchain, 0644)
+	log.Info("Certificate successfully renewed")
+
+	return string(fullchain)
+}
+
+func revokeCert(path string, cert string) {
+	// Create a client to the acme of letsencrypt.
+	log.Info("Revoking the key")
+	cli, err := letsencrypt.NewClient("https://acme-v01.api.letsencrypt.org/directory")
+	//cli, err := letsencrypt.NewClient("https://acme-staging.api.letsencrypt.org/directory")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			log.Fatal("Key can not be found")
+		}
+	}
+
+	// Find or generate the private key and register with the ACME server.
+	var key *rsa.PrivateKey
+	key, err = loadKey(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	//revoke the certificate
+	reErr := cli.RevokeCertificate(key, []byte(cert))
+	if reErr != nil {
+		log.Fatal("error while revoke:", reErr)
+	}
+	log.Print("Succesfully revoked")
+}
+
+//code addapted from :https://ericchiang.github.io/post/go-letsencrypt/ "
+func getCert(dir string, domain string) string {
+	// Create a client to the acme of letsencrypt.
+	log.Info("Requesting RSA keys")
+	cli, err := letsencrypt.NewClient("https://acme-v01.api.letsencrypt.org/directory")
+	//cli, err := letsencrypt.NewClient("https://acme-staging.api.letsencrypt.org/directory")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if _, err = os.Stat("/home/" + dir + "/cert/" + domain); os.IsNotExist(err) {
+		os.MkdirAll("/home/"+dir+"/cert/"+domain, 0777)
+	}
+
+	// Find or generate the private key to register on the ACME server.
+	var key *rsa.PrivateKey
+	key, err = loadKey("/home/" + dir + "/cert/" + domain + "/registerkey.pem")
+	if err != nil {
+		log.Print("new generate RSA key for register , keep it safe")
+		key, err = rsa.GenerateKey(rand.Reader, 2048)
+		if err != nil {
+			log.Fatal(err)
+		}
+		ioutil.WriteFile("/home/"+dir+"/cert/"+domain+"/registerkey.pem", pem.EncodeToMemory(&pem.Block{Bytes: x509.MarshalPKCS1PrivateKey(key), Type: "RSA PRIVATE KEY"}), 0644)
+	}
+
+	//register to the ACME server
+	if _, err := cli.NewRegistration(key); err != nil {
+		log.Fatal(err)
+	}
+
+	log.Print("Registration successful!")
```

Comment:
`log.Print` should in fact be called `log.Debug`, as it's only used for debugging ;) Use `log.Info` here.